src/contexts/price-oracle.context.tsx

```diff
@@ -1,29 +1,47 @@
-import { BigNumber } from "ethers";
+import { formatUnits } from "ethers/lib/utils";
 import { createContext, FC, useContext, useEffect, useState } from "react";
 
 import { useEnvContext } from "src/contexts/env.context";
+import { useGlobalContext } from "src/contexts/global.context";
 import { useProvidersContext } from "src/contexts/providers.context";
+import { Currency, FiatExchangeRates } from "src/domain";
 import { PriceOracle, PriceOracle__factory } from "src/types/contracts/price-oracle";
+import { getFiatExchangeRates } from "src/adapters/fiat-exchange-rates";
+import { USDT_DECIMALS } from "src/constants";
+import * as storage from "src/adapters/local-storage";
 
 interface PriceOracleContext {
-  getTokenPrice: (tokenAddress: string) => Promise<BigNumber>;
+  preferredCurrency?: Currency;
+  changePreferredCurrency: (currency: Currency) => void;
+  getTokenPrice: (tokenAddress: string, preferredCurrency: Currency) => Promise<number>;
 }
 
 const priceOracleContextNotReadyErrorMsg = "The price oracle context is not yet ready";
 
 const priceOracleContext = createContext<PriceOracleContext>({
+  changePreferredCurrency: () => {
+    console.error(priceOracleContextNotReadyErrorMsg);
+  },
   getTokenPrice: () => {
     console.error(priceOracleContextNotReadyErrorMsg);
-    return Promise.resolve(BigNumber.from(0));
+    return Promise.resolve(0);
```

Comment:
We should improve the pattern

```typescript
const fooNotReadyErrorMsg = "Foo context is not yet ready";
```
Instead of logging an error and returning a `Promise.resolve()`, in some cases with an arbitrary value, we should:

```typescript
return Promise.reject(fooNotReadyErrorMsg);
```

At the end of the day, if the context is not yet ready we should not resolve the promise.

If you agree, feel free to update this in all the contexts. I can also do it in my current task.

Reply:
Right, done!